src/core/context/examples.rs

```diff
@@ -0,0 +1,206 @@
+        "initialize_network",
+        "load_configuration",
+        "setup_routing",
+        "register_services",
+    ];
+    
+    for operation in startup_operations {
+        // Use run_or_throw for critical startup operations
+        // If any fails, the program terminates
+        ctx.run_or_throw(simulate_startup_operation(operation)).await;
+        tracing::info!("Completed startup operation: {}", operation);
+    }
+    
+    tracing::info!("Server startup completed successfully");
+    Ok(())
+}
+
+/// Example: Hierarchical cancellation
+pub async fn hierarchical_cancellation_example(parent_span: &Span) -> Result<()> {
+    let root_ctx = CancelableContext::new(parent_span);
+    let service1_ctx = root_ctx.child();
+    let service2_ctx = root_ctx.child();
+    
+    // Spawn multiple services
+    let service1_handle = tokio::spawn({
+        let ctx = service1_ctx.clone();
+        async move {
+            ctx.run(async {
+                tracing::info!("Service 1 starting");
+                sleep(Duration::from_secs(2)).await;
+                tracing::info!("Service 1 completed");
+                Ok(())
+            }).await
+        }
+    });
+    
+    let service2_handle = tokio::spawn({
+        let ctx = service2_ctx.clone();
+        async move {
+            ctx.run(async {
+                tracing::info!("Service 2 starting");
+                sleep(Duration::from_secs(2)).await;
+                tracing::info!("Service 2 completed");
+                Ok(())
+            }).await
+        }
+    });
+    
+    // Cancel root after short time - should cancel all services
+    tokio::spawn({
+        let ctx = root_ctx.clone();
+        async move {
+            sleep(Duration::from_millis(50)).await;
+            tracing::info!("Shutting down all services");
+            ctx.cancel();
+        }
+    });
+    
+    let (result1, result2) = tokio::join!(service1_handle, service2_handle);
+    
+    match result1 {
+        Ok(Ok(())) => tracing::info!("Service 1 completed"),
+        Ok(Err(e)) => tracing::info!("Service 1 cancelled: {}", e),
+        Err(e) => tracing::error!("Service 1 panicked: {}", e),
+    }
+    
+    match result2 {
+        Ok(Ok(())) => tracing::info!("Service 2 completed"),
+        Ok(Err(e)) => tracing::info!("Service 2 cancelled: {}", e),
+        Err(e) => tracing::error!("Service 2 panicked: {}", e),
+    }
+    
+    Ok(())
+}
+
+/// Example: Error propagation pattern
+pub async fn error_propagation_example(parent_span: &Span) -> Result<()> {
+    let root_ctx = CancelableContext::new(parent_span);
+    let child_ctx = root_ctx.child();
+    let _grandchild_ctx = child_ctx.child();
+    
+    // Simulate a critical error in deeply nested operation
+    let critical_error = anyhow!("Critical database connection failed");
+    
+    // In a real scenario, uncommenting the line below would terminate the program:
+    // grandchild_ctx.throw_irrecoverable(critical_error);
```

Comment:
Please use `assert_panic` here.

Reply:
Per our conversation, I have deleted examples.rs as there are already enough tests in context_test.rs 
https://github.com/thep2p/skipgraph-rust/pull/65/commits/003416b42c6aa7e4d3f7af2d8cca66de341dbc96